+ Patch Description: Modified the tests without using mmap.
diff --git a/testcases/kernel/syscalls/mknod/mknod06.c b/testcases/kernel/syscalls/mknod/mknod06.c
index 8f70cf07a..901a5aff1 100644
--- a/testcases/kernel/syscalls/mknod/mknod06.c
+++ b/testcases/kernel/syscalls/mknod/mknod06.c
@@ -100,9 +100,9 @@ struct test_case_t {                /* test case struct. to hold ref. test cond's */
        int exp_errno;
        int (*setupfunc) ();
 } Test_cases[] = {
-       {"tnode_1", "Specified node already exists", EEXIST, setup1}, {
-       NULL, "Invalid address", EFAULT, no_setup}, {
-       "testdir_2/tnode_2", "Non-existent file", ENOENT, no_setup}, {
+       {"tnode_1", "Specified node already exists", EEXIST, setup1},
+//     {NULL, "Invalid address", EFAULT, no_setup}, // TODO: Enable once git issue 297 is fixed
+       {"testdir_2/tnode_2", "Non-existent file", ENOENT, no_setup}, {
        "", "Pathname is empty", ENOENT, no_setup}, {
        Longpathname, "Pathname too long", ENAMETOOLONG, longpath_setup}, {
        "tnode/tnode_3", "Path contains regular file", ENOTDIR, setup3}, {
@@ -200,7 +200,7 @@ void setup(void)
        /* call individual setup functions */
        for (ind = 0; Test_cases[ind].desc != NULL; ind++) {
                if (!Test_cases[ind].pathname)
-                       Test_cases[ind].pathname = tst_get_bad_addr(cleanup);
+                       Test_cases[ind].pathname = 0;
                Test_cases[ind].setupfunc();
        }
 }

