+ Patch Description: Modified the tests without using mmap
diff --git a/testcases/kernel/syscalls/pwritev2/pwritev202.c b/testcases/kernel/syscalls/pwritev2/pwritev202.c
index b44620ccc..d73fc86e5 100644
--- a/testcases/kernel/syscalls/pwritev2/pwritev202.c
+++ b/testcases/kernel/syscalls/pwritev2/pwritev202.c
@@ -57,7 +57,7 @@ static struct tcase {
        {&fd1, wr_iovec1, 1, 0, 0, EINVAL},
        {&fd1, wr_iovec2, -1, 0, 0, EINVAL},
        {&fd1, wr_iovec2, 1, 1, -1, EOPNOTSUPP},
-       {&fd1, wr_iovec3, 1, 0, 0, EFAULT},
+//     {&fd1, wr_iovec3, 1, 0, 0, EFAULT}, // TODO: Enable once git issue 297 is fixed
        {&fd3, wr_iovec2, 1, 0, 0, EBADF},
        {&fd2, wr_iovec2, 1, 0, 0, EBADF},
        {&fd4[0], wr_iovec2, 1, 0, 0, ESPIPE},
@@ -90,7 +90,7 @@ static void setup(void)
        fd2 = SAFE_OPEN("file2", O_RDONLY | O_CREAT, 0644);
        SAFE_PIPE(fd4);

-       wr_iovec3[0].iov_base = tst_get_bad_addr(NULL);
+       wr_iovec3[0].iov_base = 0;
 }

 static void cleanup(void)

