+ Patch Description: Modified tests without mmap.
diff --git a/testcases/kernel/syscalls/chroot/chroot03.c b/testcases/kernel/syscalls/chroot/chroot03.c
index b904e4ac9..6c6b5380e 100644
--- a/testcases/kernel/syscalls/chroot/chroot03.c
+++ b/testcases/kernel/syscalls/chroot/chroot03.c
@@ -79,14 +79,14 @@ struct test_case_t {
             */
        {
        nonexistent_dir, ENOENT},
-#if !defined(UCLINUX)
+//#if !defined(UCLINUX) // TODO: Enable once git issue 297 is fixed
            /*
             * attempt to chroot to a path pointing to an invalid address
             * and expect EFAULT as errno
             */
-       {
-       (char *)-1, EFAULT},
-#endif
+//     {
+//     (char *)-1, EFAULT},
+//#endif
        {symbolic_dir, ELOOP}
 };

@@ -146,12 +146,9 @@ static void setup(void)
        fd = SAFE_CREAT(cleanup, fname, 0777);

 #if !defined(UCLINUX)
-       bad_addr = mmap(0, 1, PROT_NONE,
-                       MAP_PRIVATE_EXCEPT_UCLINUX | MAP_ANONYMOUS, 0, 0);
-       if (bad_addr == MAP_FAILED)
-               tst_brkm(TBROK, cleanup, "mmap failed");
+       bad_addr = 0;

-       TC[3].dir = bad_addr;
+//     TC[3].dir = bad_addr; // TODO: Enable once git issue 297 is fixed
 #endif
        /*
         * create two symbolic directory who point to each other to

