+ Patch Description: Modified the tests without using mmap.
diff --git a/testcases/kernel/syscalls/rmdir/rmdir02.c b/testcases/kernel/syscalls/rmdir/rmdir02.c
index cb0aec857..25738cb32 100644
--- a/testcases/kernel/syscalls/rmdir/rmdir02.c
+++ b/testcases/kernel/syscalls/rmdir/rmdir02.c
@@ -42,7 +42,7 @@ static struct testcase {
        {longpathname, ENAMETOOLONG},
        {TESTDIR2, ENOENT},
        {TESTDIR3, ENOTDIR},
-       {NULL, EFAULT},
+//     {NULL, EFAULT},
        {looppathname, ELOOP},
        {TESTDIR5, EROFS},
        {MNT_POINT, EBUSY},
@@ -62,7 +62,7 @@ static void setup(void)

        for (i = 0; i < ARRAY_SIZE(tcases); i++) {
                if (!tcases[i].dir)
-                       tcases[i].dir = tst_get_bad_addr(NULL);
+                       tcases[i].dir = 0;
        }

        /*

