+ Patch Description: Modified the tests without using mmap
diff --git a/testcases/kernel/syscalls/statx/statx03.c b/testcases/kernel/syscalls/statx/statx03.c
index c72d7fead..2b0c9a7b7 100644
--- a/testcases/kernel/syscalls/statx/statx03.c
+++ b/testcases/kernel/syscalls/statx/statx03.c
@@ -55,8 +55,8 @@ static struct test_case {
        {.dfd = -1, .filename = &test_fname, .flag = 0,
         .mask = 0, .errnum = EBADF},

-       {.dfd = AT_FDCWD, .filename = &efault_fname, .flag = 0,
-        .mask = 0, .errnum = EFAULT},
+//     {.dfd = AT_FDCWD, .filename = &efault_fname, .flag = 0, // TODO: Enable once git issue 297 is fixed
+//      .mask = 0, .errnum = EFAULT},

        {.dfd = AT_FDCWD, .filename = &test_fname, .flag = -1,
         .mask = 0, .errnum = EINVAL},
@@ -106,7 +106,7 @@ static void setup(void)
        memset(long_pathname, '@', sizeof(long_pathname));
        long_pathname[sizeof(long_pathname) - 1] = 0;

-       efault_fname = tst_get_bad_addr(NULL);
+       efault_fname = 0;
 }

 static struct tst_test test = {

