+ Patch Description: Modified the tests without using mmap.
diff --git a/testcases/kernel/syscalls/stat/stat03.c b/testcases/kernel/syscalls/stat/stat03.c
index 20d4c772f..dcf9f8d71 100644
--- a/testcases/kernel/syscalls/stat/stat03.c
+++ b/testcases/kernel/syscalls/stat/stat03.c
@@ -33,7 +33,7 @@ static struct tcase{
        int exp_errno;
 } TC[] = {
        {TST_EACCES_FILE, EACCES},
-       {NULL, EFAULT},
+//     {NULL, EFAULT}, // TODO: Enable once git issue 297 is fixed
        {long_dir, ENAMETOOLONG},
        {TST_ENOENT, ENOENT},
        {TST_ENOTDIR_DIR, ENOTDIR},
@@ -73,7 +73,7 @@ static void setup(void)

        for (i = 0; i < ARRAY_SIZE(TC); i++) {
                if (TC[i].exp_errno == EFAULT)
-                       TC[i].pathname = tst_get_bad_addr(NULL);
+                       TC[i].pathname = 0;
        }

        SAFE_TOUCH(TST_ENOTDIR_FILE, DIR_MODE, NULL);
