diff -urN protobuf-c-old/configure protobuf-c/configure
--- protobuf-c-old/configure	2018-08-14 05:25:28.000000000 +0100
+++ protobuf-c/configure	2019-03-21 16:45:19.674033764 +0000
@@ -17613,26 +17613,6 @@
 	proto3_supported=yes
 fi
 
-  save_CPPFLAGS="$CPPFLAGS"
-  CPPFLAGS="$save_CPPFLAGS $protobuf_CFLAGS"
-
-for ac_header in google/protobuf/compiler/command_line_interface.h
-do :
-  ac_fn_cxx_check_header_mongrel "$LINENO" "google/protobuf/compiler/command_line_interface.h" "ac_cv_header_google_protobuf_compiler_command_line_interface_h" "$ac_includes_default"
-if test "x$ac_cv_header_google_protobuf_compiler_command_line_interface_h" = xyes; then :
-  cat >>confdefs.h <<_ACEOF
-#define HAVE_GOOGLE_PROTOBUF_COMPILER_COMMAND_LINE_INTERFACE_H 1
-_ACEOF
-
-else
-  as_fn_error $? "required protobuf header file not found" "$LINENO" 5
-fi
-
-done
-
-  CPPFLAGS="$save_CPPFLAGS"
-
-
   # Extract the first word of "protoc", so it can be a program name with args.
 set dummy protoc; ac_word=$2
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
diff -urN protobuf-c-old/configure.ac protobuf-c/configure.ac
--- protobuf-c-old/configure.ac	2018-08-14 05:24:20.000000000 +0100
+++ protobuf-c/configure.ac	2019-03-21 16:35:43.172970003 +0000
@@ -83,13 +83,6 @@
     [PKG_CHECK_MODULES([protobuf], [protobuf >= 2.6.0])]
   )
 
-  save_CPPFLAGS="$CPPFLAGS"
-  CPPFLAGS="$save_CPPFLAGS $protobuf_CFLAGS"
-  AC_CHECK_HEADERS([google/protobuf/compiler/command_line_interface.h],
-                   [],
-                   [AC_MSG_ERROR([required protobuf header file not found])])
-  CPPFLAGS="$save_CPPFLAGS"
-
   AC_ARG_VAR([PROTOC], [protobuf compiler command])
   AC_PATH_PROG([PROTOC], [protoc], [],
                [`$PKG_CONFIG --variable=exec_prefix protobuf`/bin:$PATH])
diff -urN protobuf-c-old/Makefile.am protobuf-c/Makefile.am
--- protobuf-c-old/Makefile.am	2018-08-14 03:11:41.000000000 +0100
+++ protobuf-c/Makefile.am	2019-03-21 16:35:14.833310969 +0000
@@ -98,8 +98,7 @@
 	protoc-c/c_service.cc \
 	protoc-c/c_service.h \
 	protoc-c/c_string_field.cc \
-	protoc-c/c_string_field.h \
-	protoc-c/main.cc
+	protoc-c/c_string_field.h
 protoc_c_protoc_gen_c_CXXFLAGS = \
 	$(AM_CXXFLAGS) \
 	$(protobuf_CFLAGS)
diff -urN protobuf-c-old/protoc-c/main.cc protobuf-c/protoc-c/main.cc
--- protobuf-c-old/protoc-c/main.cc	2017-08-05 22:43:57.000000000 +0100
+++ protobuf-c/protoc-c/main.cc	1970-01-01 01:00:00.000000000 +0100
@@ -1,22 +0,0 @@
-#include <string>
-
-#include <google/protobuf/compiler/plugin.h>
-#include <google/protobuf/compiler/command_line_interface.h>
-#include <protoc-c/c_generator.h>
-
-int main(int argc, char* argv[]) {
-  google::protobuf::compiler::c::CGenerator c_generator;
-
-  std::string invocation_name = argv[0];
-  std::string invocation_basename = invocation_name.substr(invocation_name.find_last_of("/") + 1);
-  const std::string legacy_name = "protoc-c";
-
-  if (invocation_basename == legacy_name) {
-    google::protobuf::compiler::CommandLineInterface cli;
-    cli.RegisterGenerator("--c_out", &c_generator, "Generate C/H files.");
-    cli.SetVersionInfo(PACKAGE_STRING);
-    return cli.Run(argc, argv);
-  }
-
-  return google::protobuf::compiler::PluginMain(argc, argv, &c_generator);
-}
